Affected strategy analyze .class files when target dir not ignored

##### Issue Overview

Affected strategy analyze `.class` files when target dir not ignored from git (`.gitignore`) and this causes an exception: 

```
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
[ERROR] Unable to analyze source file /Users/alex/git/stdemo/target/classes/org/superbiz/A.class
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:665)
```
##### Expected Behaviour

`.class` files should be ignored.

##### Current Behaviour

Exception is thrown.

##### Steps To Reproduce
1. Clone https://github.com/lordofthejars/stdemo
2. Edit `.gitignore` and comment `target/` line using `#`
3. Run `mvn clean test -Dsmart.testing="affected"`

--- a/core/src/main/java/org/arquillian/smart/testing/api/TestVerifier.java
+++ b/core/src/main/java/org/arquillian/smart/testing/api/TestVerifier.java
@@ -16,6 +16,11 @@ public interface TestVerifier {
      * @return Whether the given path to a resource represents a test class or not
      */
     default boolean isTest(Path resource) {
+
+        if (!isJavaFile(resource)) {
+            return false;
+        }
+
         final String className = new ClassNameExtractor().extractFullyQualifiedName(resource);
         return isTest(className);
     }
@@ -27,9 +32,15 @@ public interface TestVerifier {
      * @return Whether the given path to a resource represents a non-test class or not
      */
     default boolean isCore(Path resource) {
+        if (!isJavaFile(resource)) {
+            return false;
+        }
         return !isTest(resource);
     }
 
+    default boolean isJavaFile(Path file) {
+        return file.toString().endsWith(".java");
+    }
 
     /**
      * Check whether the given {@link TestSelection} instance represents a test class or not

